packages/host/app/components/worker-render.gts

```diff
@@ -1,67 +1,65 @@
 import Component from '@glimmer/component';
+// import Modifier from 'ember-modifier';
 import { service } from '@ember/service';
 import Render from './render';
-import Modifier from 'ember-modifier';
 //@ts-ignore glint does not think this is consumed-but it is consumed in the template
 import { hash } from '@ember/helper';
-import type WorkerRenderer from '../services/worker-renderer';
-import type { Card, Format } from 'https://cardstack.com/base/card-api';
+import Serializer from '@simple-dom/serializer';
+import voidMap from '@simple-dom/void-map';
+import type IndexerService from '../services/indexer-service';
+import type { SimpleDocument } from '@simple-dom/interface';
 
 export default class WorkerRender extends Component {
   <template>
-    {{#if this.needsRender}}
-      <div {{HTMLSnapshot this.card this.format}} class="worker-render">
-        <Render @card={{this.card}} @format={{this.format}} @opts={{hash disableShadowDOM=true}}/>
+    {{#if this.indexerService.card}}
+      <div class="worker-render">
+        <Render @card={{this.indexerService.card}} @format="isolated" @opts={{hash disableShadowDOM=true}}/>
       </div>
     {{/if}}
   </template>
 
-  @service declare workerRenderer: WorkerRenderer;
+  @service declare indexerService: IndexerService;
+  document = service('-document') as unknown as SimpleDocument;
+  // @service('-document') document;
 
-  get needsRender() {
-    return this.workerRenderer.card != null && this.workerRenderer.format != null;
-  }
-  get card() {
-    if (!this.workerRenderer.card) {
-      throw new Error('bug: should never be here');
-    }
-    return this.workerRenderer.card;
-  }
-  get format() {
-    if (!this.workerRenderer.format) {
-      throw new Error('bug: should never be here');
-    }
-    return this.workerRenderer.format;
+  constructor(owner: unknown, args: any) {
+    super(owner, args);
+    debugger;
+    // TODO hopefully this.document contains this component (???)
+    let serializer = new Serializer(voidMap);
+    let html = serializer.serialize(this.document); // TODO use simple DOM to get this component's element instead of using whole doc
+    // TODO use our tokens to trim out our card's HTML from the document
+    this.indexerService.captureSnapshot(html);
+  
+    // TODO when this.args.card changes, does that trigger a new
+    // WorkerRender component? if not how weill we hook into the 
+    // change to be able to call captureHTML again?
   }
 }
+    
```

Comment:
@ef4 I'm ready for a discussion around this. this particular component is the component that drives the card prerendering. This component is included in the application.hbs template, and uses the IndexerService's tracked `card` property to render the current card. I've setup the `CurrentRun` so that as each card is indexed it calls the IndexerService.viewCard, which loads the card and sets it's tracked `card` property such that this component will pick it up and render the card. In fastboot this component never seems to even instantiate--the debugger in the constructor never pauses.